xandikos/webdav.py

```diff
@@ -1485,7 +1485,7 @@ def _get_resources_by_hrefs(backend, environ, hrefs):
     script_name = environ["SCRIPT_NAME"]
     # TODO(jelmer): Bulk query hrefs in a more efficient manner
     for href in hrefs:
-        if not href.startswith(script_name):
+        if not href or not href.startswith(script_name):
```

Comment:
With your example, shouldn't href be empty rather than None?

Reply:
Seems like the xml parser returns None instead of `""`

```py
>>> from defusedxml.ElementTree import fromstring as xmlparse
>>> print(xmlparse("<a>foo</a>").text)
foo
>>> print(xmlparse("<a></a>").text)
None
```